sdk/python/kfp/compiler/compiler.py

```diff
@@ -304,8 +304,8 @@ def _build_conventional_artifact(name, path):
     # for the artifact output when default artifact repository is configured,
     # this part needs to be updated to use the default artifact repository.
     output_artifacts = []
-    output_artifacts.append(_build_conventional_artifact('mlpipeline-ui-metadata', '/mlpipeline-ui-metadata.json'))
-    output_artifacts.append(_build_conventional_artifact('mlpipeline-metrics', '/mlpipeline-metrics.json'))
+    output_artifacts.append(_build_conventional_artifact('mlpipeline-ui-metadata', op.mlpipeline_ui_metadata_path))
```

Comment:
https://github.com/kubeflow/pipelines/pull/998#issuecomment-474647713
This could be:

```
op._output_artifact_paths.setdefault('mlpipeline-ui-metadata', '/mlpipeline-ui-metadata.json')
op._output_artifact_paths.setdefault('mlpipeline-metrics', '/mlpipeline-metrics.json')
output_artifacts = [_build_conventional_artifact(name, path) for name, path in op._output_artifact_paths.items()]
```

Reply:
Thanks for the suggestion. I updated it in my latest commit